### PR TITLE
Changes for #75, CLI-support for user controlled in-memory copy operation

### PR DESCRIPTION
**Pull request by ksic8**

### CODE REVIEW

File: weather_mv/loader_pipeline/sinks.py [hunk 1]
```diff
@@ -52,34 +53,44 @@ def _make_grib_dataset_inmem(grib_ds: xr.Dataset) -> xr.Dataset:
...
     logger.warning("Assuming grib edition 1.")
     # Try with edition 1
     # Note: picking edition 1 for now as it seems to get the most data/variables for ECMWF realtime data.
-    return _make_grib_dataset_inmem(xr.open_dataset(filename, engine='cfgrib',
-                                                    backend_kwargs={'filter_by_keys': {'edition': 1}, 'indexpath': ''}))
+    return xr.open_dataset(filename, engine='cfgrib',
+                           backend_kwargs={'filter_by_keys': {'edition': 1}, 'indexpath': ''}) \
+        if disable_in_memory_copy \
+        else _make_grib_dataset_inmem(
+        xr.open_dataset(filename, engine='cfgrib', backend_kwargs={'filter_by_keys': {'edition': 1}, 'indexpath': ''}))
```
Comment (alxmrs) [inline weather_mv/loader_pipeline/sinks.py:88]: Please make this more DRY (don't repeat yourself). I notice identical functions are called in the if statement. This is fairly brittle – what if someone later updates one method but not the other?

**Review by alxmrs — CHANGES_REQUESTED** [review]

File: weather_mv/loader_pipeline/bq.py [hunk 2]
```diff
@@ -82,7 +82,8 @@ def __post_init__(self):
             )
         else:
             logger.info('Inferring schema from data.')
-            with open_dataset(self.example_uri, self.xarray_open_dataset_kwargs) as open_ds:
+            with open_dataset(self.example_uri, self.xarray_open_dataset_kwargs,
+                              self.disable_in_memory_copy) as open_ds:
```
[the remark this answers is not in the document]
Comment (ksic8) [reply weather_mv/loader_pipeline/bq.py]: Agreed. Revised code pushed for your review.
Note: We have kept a flag to disable in-memory-copy. Hence, the value passed from this function shall be `True`.

File: weather_mv/loader_pipeline/sinks.py (hunk 1 above)
Comment (ksic8) [reply weather_mv/loader_pipeline/sinks.py]: Necessary code-changes completed and submitted for your review.

File: weather_mv/loader_pipeline/bq.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (alxmrs) [reply weather_mv/loader_pipeline/bq.py]: Ah, thanks – I do mean true.

**Review by alxmrs — CHANGES_REQUESTED** [review]

File: weather_mv/loader_pipeline/bq.py [hunk 3]
```diff
@@ -82,7 +82,8 @@ def __post_init__(self):
             )
         else:
             logger.info('Inferring schema from data.')
-            with open_dataset(self.example_uri, self.xarray_open_dataset_kwargs) as open_ds:
+            with open_dataset(self.example_uri, self.xarray_open_dataset_kwargs,
+                              True) as open_ds:
```
[the remark this answers is not in the document]
Comment (ksic8) [reply weather_mv/loader_pipeline/bq.py]: Revised code committed.

File: weather_mv/loader_pipeline/sinks.py [hunk 4]
```diff
@@ -52,34 +53,43 @@ def _make_grib_dataset_inmem(grib_ds: xr.Dataset) -> xr.Dataset:
...
+            return xr.open_dataset(filename, engine='cfgrib', backend_kwargs={'indexpath': ''})
+        except ValueError as e:
+            if "multiple values for key 'edition'" not in str(e):
+                raise
+
+        logger.warning("Assuming grib edition 1.")
+        # Try with edition 1
+        # Note: picking edition 1 for now as it seems to get the most data/variables for ECMWF realtime data.
+        return xr.open_dataset(filename, engine='cfgrib',
+                               backend_kwargs={'filter_by_keys': {'edition': 1}, 'indexpath': ''})
```
[the remark this answers is not in the document]
Comment (ksic8) [reply weather_mv/loader_pipeline/sinks.py]: The said flag has been moved to `open_dataset` function.

File: weather_mv/loader_pipeline/sinks.py [hunk 5]
```diff
@@ -90,7 +100,8 @@ def open_dataset(uri: str, open_dataset_kwargs: t.Optional[t.Dict] = None) -> t.
                 shutil.copyfileobj(source_file, dest_file, DEFAULT_READ_BUFFER_SIZE)
                 dest_file.flush()
                 dest_file.seek(0)
-                xr_dataset: xr.Dataset = __open_dataset_file(dest_file.name, open_dataset_kwargs)
+                xr_dataset: xr.Dataset = __open_dataset_file(dest_file.name, open_dataset_kwargs,
```
[the remark this answers is not in the document]
Comment (ksic8) [reply weather_mv/loader_pipeline/sinks.py]: The flag has been moved to given function.

File: weather_mv/loader_pipeline/sinks.py [hunk 6]
```diff
@@ -43,18 +44,19 @@ def from_kwargs(cls, **kwargs):
 
 
 def _make_grib_dataset_inmem(grib_ds: xr.Dataset) -> xr.Dataset:
-    # Copies all the vars to in-memory to reduce disk seeks everytime a single row is processed.
-    # This also removes the need to keep the backing temp source file around.
+    """Copies all the vars to in-memory to reduce disk seeks everytime a single row is processed.
+    This also removes the need to keep the backing temp source file around."""
```
Comment (alxmrs) [inline weather_mv/loader_pipeline/sinks.py:48]:
```suggestion
    """Copies all the vars in-memory to reduce disk seeks every time a single row is processed.
    
    This also removes the need to keep the backing temp source file around.
    """
```

**Review by alxmrs — APPROVED** [review]
One minor nit with a comment; otherwise, LGTM! Please fix the last small thing before merging.

File: weather_mv/loader_pipeline/sinks.py (hunk 6 above)
Comment (ksic8) [reply weather_mv/loader_pipeline/sinks.py]: Docstring modified.

**Pull request merged**